File: ccif/src/java/other/org/himalay/ccif/FeatureParamters_1_4.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:10 EST 2016")

public  class FeatureParamters_1_4 extends   APDUBase_1Factory.APDUBase_1 { //Concrete type is FeatureParamters_1_4




    // members variables
    // header
    public APDUHeader header ;
    // length
    public short length ;
    // featureList
    public ArrayList<flist> featureList ;

    public FeatureParamters_1_4 () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize length

        // Initialize featureList
        featureList= new ArrayList<flist>();
        featureList.setMemberSize(0);
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        preRead();
        int retVal= 0;
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read featureList
        for (; istream.available() > 0 ; ) {
            flist temp;
            temp = new flist();
            retVal += temp.read(istream);
            featureList.add(temp);
        }

        postRead();
        return retVal;
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read header
        retVal += header.read(istream);
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read featureList
        for (; istream.available() > 0 ; ) {
            flist temp;
            temp = new flist();
            retVal += temp.read(istream);
            featureList.add(temp);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for header **/
        }

        {   /** fix dependent sizes for featureList**/
        }

        // write header
        if (header!=null)retVal +=header.write(ostream);
        // write length
        ostream.writeByte(length);
        retVal +=1;
        // write featureList
        {
            ArrayList<flist> temp1 = featureList;
            for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++) {
                flist temp2    =    temp1.get(iIdx);
                if (temp2!=null)retVal +=temp2.write(ostream);
            }
        }
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("FeatureParamters_1_4\n");
        dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {
            dc.indent();
            dc.getPs().println("header") ;
            retVal +=header.dump(dc);
        }
        // write length
        dc.indent();
        dc.getPs().println("length="+length+"(0x"+ Integer.toHexString(length) + ")" );
        // write featureList
        {
            ArrayList<flist> temp1 = featureList;
            for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++) {
                flist element    = temp1.get(iIdx);
                dc.indent();
                dc.getPs().println(iIdx);
                if ( element != null ) {
                    dc.indent();
                    dc.getPs().println("element") ;
                    retVal +=element.dump(dc);
                }
            }
        }
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}


    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for length
    //public short getLength()
    //{
    //    return length ;
    //}


    // Setter for length
    //public void setLength(short val)
    //{
    //    this.length= val;
    //}
    // Getter for featureList
    //public ArrayList<flist> getFeatureList()
    //{
    //    return featureList ;
    //}


    // Setter for featureList
    //public void setFeatureList(ArrayList<flist> val)
    //{
    //    this.featureList= val;
    //}


    public int addToFeatureList(flist val)
    {
        featureList.add(val);
        return featureList.size();
    }


    public int removeFromFeatureList(flist val)
    {
        featureList.remove(val);
        return featureList.size();
    }


    public int removeNthFromFeatureList(int idx)
    {
        featureList.remove(idx);
        return featureList.size();
    }


    public int emptyFeatureList(int idx)
    {
        featureList.clear();
        return featureList.size();
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }


    public void setHeader(APDUHeader header)
    {
        this.header= header;
    }

    public APDUHeader getHeader()
    {
        return this.header;
    }

}

// End of code

File: libccif/src/bp/FeatureParamters_1_4.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:18 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "FeatureParamters_1_4.h"

#ifdef org_himalay_ccif_FeatureParamters_1_4__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
FeatureParamters_1_4::FeatureParamters_1_4() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // length
    length= (ui8)0;
    // featureList
    featureList.setMemberSize(0);
}
// Destructor
FeatureParamters_1_4::~FeatureParamters_1_4() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // length

    // featureList
    {
#ifndef org_himalay_ccif_flist_USE_SMART_PTR
        featureList.deleteMembers();
#endif
    }
}

int FeatureParamters_1_4::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // featureList
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_flist__PTR_TYPE temp;
        org_himalay_ccif_flist__NEW( temp);
        retVal += temp->read(istream);
        featureList.add(temp);
    }
    return retVal;
}

int FeatureParamters_1_4::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // read featureList
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_flist__PTR_TYPE temp;
        org_himalay_ccif_flist__NEW( temp);
        retVal += temp->read(istream);
        featureList.add(temp);
    }

    return retVal;
}

int FeatureParamters_1_4::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }

    {   /** fix dependent sizes for featureList**/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write length
    ostream.writeUI8(length);
    retVal +=1;
    // write featureList
{   for (ArrayList_iterator(flist) it=  featureList.getIterator() ; it.hasNext();) {
            {
                retVal +=it.get()->write(ostream);
            }
        }
    }

    return retVal;
}

int FeatureParamters_1_4::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("FeatureParamters_1_4\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write length
    dc.indent();
    dc.getPs().print("length=");
    dc.getPs().println((long)length);
    // write featureList
{   for (ArrayList_iterator(flist) it=  featureList.getIterator() ; it.hasNext();) {
            {
                dc.indent();
                dc.getPs().println("element");
                retVal +=it.get()->dump(dc);
            }
        }
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE FeatureParamters_1_4::getHeader()
//{
//return this->header;
//}

// Setter for header
//void FeatureParamters_1_4::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for length
//ui8 FeatureParamters_1_4::getLength()
//{
//return this->length;
//}

// Setter for length
//void FeatureParamters_1_4::setLength(ui8 val)
//{
//this->length= val;
//}

// Getter for featureList
//ArrayList(flist) FeatureParamters_1_4::getFeatureList()
//{
//return this->featureList;
//}

// Setter for featureList
//void FeatureParamters_1_4::setFeatureList(ArrayList(flist) val)
//{
//this->featureList= val;
//}

int FeatureParamters_1_4::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void FeatureParamters_1_4::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE header)
{
    this->header= header;
}

org_himalay_ccif_APDUHeader__PTR_TYPE FeatureParamters_1_4::getHeader()
{
    return this->header;
}

#ifdef org_himalay_ccif_FeatureParamters_1_4__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::FeatureParamters_1_4* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::FeatureParamters_1_4* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code